src/features/cookie.js

```diff
@@ -0,0 +1,215 @@
+    } catch {
+        framingOrigin = globalThis.document.referrer
+    }
+
+    // Not supported in Firefox
+    if ('ancestorOrigins' in globalThis.location && globalThis.location.ancestorOrigins.length) {
+        // ancestorOrigins is reverse order, with the last item being the top frame
+        framingOrigin = globalThis.location.ancestorOrigins.item(globalThis.location.ancestorOrigins.length - 1)
+    }
+    return framingOrigin
+}
+
+// Initial cookie policy pre init
+let cookiePolicy = {
+    debug: false,
+    isFrame: isBeingFramed(),
+    isTracker: false,
+    shouldBlock: true,
+    shouldBlockTrackerCookie: true,
+    shouldBlockNonTrackerCookie: true,
+    isThirdParty: isThirdParty(),
+    tabRegisteredDomain: getTabOrigin(),
+    policy: {
+        threshold: 864000, // 10 days
+        maxAge: 864000 // 10 days
+    }
+}
+
+let loadedPolicyResolve
+// Listen for a message from the content script which will configure the policy for this context
+const trackerHosts = new Set()
+
+/**
+ * @param {'ignore' | 'block' | 'restrict'} action
+ * @param {string} reason
+ * @param {any} ctx
+ */
+function debugHelper (action, reason, ctx) {
+    cookiePolicy.debug && postDebugMessage('jscookie', {
+        action,
+        reason,
+        stack: ctx.stack,
+        documentUrl: globalThis.document.location.href,
+        scriptOrigins: [...ctx.scriptOrigins],
+        value: ctx.value
+    })
+}
+
+function shouldBlockTrackingCookie () {
+    return cookiePolicy.isFrame && cookiePolicy.shouldBlock && cookiePolicy.isTracker && cookiePolicy.shouldBlockTrackerCookie && cookiePolicy.isThirdParty
+}
+
+function shouldBlockNonTrackingCookie () {
+    return cookiePolicy.isFrame && cookiePolicy.shouldBlock && !cookiePolicy.isTracker && cookiePolicy.shouldBlockNonTrackerCookie && cookiePolicy.isThirdParty
+}
+
+export function load (args) {
+    trackerHosts.clear()
+
+    // The cookie policy is injected into every frame immediately so that no cookie will
+    // be missed.
+    const document = globalThis.document
+    const cookieSetter = Object.getOwnPropertyDescriptor(globalThis.Document.prototype, 'cookie').set
+    const cookieGetter = Object.getOwnPropertyDescriptor(globalThis.Document.prototype, 'cookie').get
+
+    const loadPolicy = new Promise((resolve) => {
+        loadedPolicyResolve = resolve
+    })
+    // Create the then callback now - this ensures that Promise.prototype.then changes won't break
+    // this call.
+    const loadPolicyThen = loadPolicy.then.bind(loadPolicy)
+
+    function getCookiePolicy () {
+        const stack = getStack()
+        const scriptOrigins = getStackTraceOrigins(stack)
+        const getCookieContext = {
+            stack,
+            scriptOrigins,
+            value: 'getter'
+        }
+
+        if (shouldBlockTrackingCookie() || shouldBlockNonTrackingCookie()) {
+            if (cookiePolicy.debug) {
+                debugHelper('block', 'frame', getCookieContext)
```

Comment:
Should this say 3p frame as well like on line 135?

Reply:
Cleared this up anyway also add an ignore case :)